### ISSUE TITLE
[winui3]How to get parent HWND of TextBox in WinUI3? How to use ImmAssociateContext in WinUI3?

### ISSUE THREAD

**Issue by TanWei** [microsoft/microsoft-ui-xaml#7275]
I am using winui3 in cpp.
I want to disable IME(Input Method) when using TextBox.
One way I know is using
~~~
 ImmAssociateContext(hwnd, NULL);
~~~
The problem is thar how I get parent HWND of  TextBox?

or is there another way to disable IME?

**Comment by castorix** [comment]
There is only one HWND (3 with the window for XAML rendering and the window for the Caption), 
which is, from MainWindow, `hWnd = WinRT.Interop.WindowNative.GetWindowHandle(this);`
But _ImmAssociateContext_ does not work for me (Windows 10 21H1)
(same thing with _ImmDisableIME_ or _ImmDisableLegacyIME_, which work before main window creation in C++/win32)

**Comment by TanWei** [comment]
> There is only one HWND (3 with the window for XAML rendering and the window for the Caption), which is, from MainWindow, `hWnd = WinRT.Interop.WindowNative.GetWindowHandle(this);` But _ImmAssociateContext_ does not work for me (Windows 10 21H1) (same thing with _ImmDisableIME_ or _ImmDisableLegacyIME_, which work before main window creation in C++/Win32)

Thanks for your reply.
In my cognition, IME must be attached to a window to display, so that function 'ImmAssociateContext(hwnd, NULL);' should be able to disable it.Unless the HWND does not match.
I also tried calling the VisualTreeHelper::GetParent(TextBOX()); method in a loop to get the parent window of the TextBox, but the result is empty in the first loop.
I noticed that PassWordBox doesn't pop up the IME, can I see how PassWordBox does it? Is his implementation open source and Where can I find his code?

**Comment by seven-mile** [comment]
> > There is only one HWND (3 with the window for XAML rendering and the window for the Caption), which is, from MainWindow, `hWnd = WinRT.Interop.WindowNative.GetWindowHandle(this);` But _ImmAssociateContext_ does not work for me (Windows 10 21H1) (same thing with _ImmDisableIME_ or _ImmDisableLegacyIME_, which work before main window creation in C++/Win32)
> 
> Thanks for your reply. In my cognition, IME must be attached to a window to display, so that function 'ImmAssociateContext(hwnd, NULL);' should be able to disable it.Unless the HWND does not match. I also tried calling the VisualTreeHelper::GetParent(TextBOX()); method in a loop to get the parent window of the TextBox, but the result is empty in the first loop. I noticed that PassWordBox doesn't pop up the IME, can I see how PassWordBox does it? Is his implementation open source and Where can I find his code?

No, it's not open source yet. The UI library itself have full access to the control implementation. So they can achieve it in any way. But you're not. Probably this should be a feature request that TextBox provides a property like `IMEEnabled` (or external `TextService.IMEEnabled`? I don't know).

**Comment by castorix** [comment]
I posted an answer on MS forums with _ShowReadingWindow_  but you did not read it...

**Comment by TanWei** [comment]
Thanks for [castorix],I have resolved this problem by using ITfThreadMgr::Deactivate.
If anyone else needs help, you can check out：https://docs.microsoft.com/en-us/answers/questions/900827/winui3-how-to-disable-input-methods-when-using-tex.html
